src/plugins/controls/public/control_group/editor/data_control_editor_tools.ts

```diff
@@ -6,13 +6,20 @@
  * Side Public License, v 1.
  */
 
+import { memoize } from 'lodash';
+
 import { IFieldSubTypeMulti } from '@kbn/es-query';
 import { DataView } from '@kbn/data-views-plugin/common';
 
 import { pluginServices } from '../../services';
 import { DataControlFieldRegistry, IEditableControlFactory } from '../../types';
 
-const dataControlFieldRegistryCache: { [key: string]: DataControlFieldRegistry } = {};
+export const getDataControlFieldRegistry = memoize(
+  async (dataView: DataView) => {
+    return await loadFieldRegistryFromDataViewId(dataView);
+  },
+  (dataView: DataView) => [dataView.id, JSON.stringify(dataView.fields.getAll())].join('|')
+);
```

Comment:
With the old "fake" cache, because it relied solely on data view IDs, adding a runtime field required a hard refresh in order to get this new field to show up in the field list for creating/editing controls. By using Lodash's `memoize` instead, I can force the cache to update when the fields change, so a hard refresh is no longer required when runtime fields are added.